lib/Dialect/Torch/Transforms/SimplifyShapeCalculations.cpp

```diff
@@ -346,12 +346,18 @@ static void refineShapeCalculateResult(ShapeCalculateOp op, int resultNum,
   if (!updatedType || updatedType == originalResultType)
     return;
 
+  // Since we may create new ops in the following loop body, save a copy of the
+  // original uses to prevent the iterator from being invalidated.
+  auto originalUses = llvm::to_vector<6>(
+      llvm::map_range(result.getUses(), [](OpOperand &use) { return &use; }));
+
   // Update all the uses of the result type to the new type, if possible. Insert
   // a TensorStaticInfoCastOp for any users that might require the exact
   // previous type.
   Value originalTypedValue;
-  for (OpOperand &use : result.getUses()) {
-    if (use.getOwner()
+
+  for (OpOperand *use : originalUses) {
```

Comment:
I think llvm::make_early_inc_range would fix this too.

Reply:
Ah nice!  It's now just a one-line change.  Thanks!